### ISSUE TITLE
Second `yield from` is not called from a data provider

### ISSUE THREAD

**Issue by markbook2** [sebastianbergmann/phpunit#2693]
| Q                   | A
| --------------------| ---------------
| PHPUnit version     | PHPUnit 6.2-gd3180f13e 
| PHP version         | 7.1.5
| Installation Method | Composer, dev-master

A data provider can return data from generator, however if a generator returns the same key, the result will not be produced. The following code:
```php
class MyTest extends TestCase {
    public function dataFor404StatusCode() {
        yield from $this->yieldUris('foo/hello');
        yield from $this->yieldUris('bar/world');
    }

    /**
     * @dataProvider dataFor404StatusCode
     */
    public function test404StatusCode(string $uri) {
        var_dump($uri);
    }

    private function yieldUris(string $uriWithoutProtocol): \Generator {
        yield ['http://' . $uriWithoutProtocol];
        yield ['http://www.' . $uriWithoutProtocol];
    }
}
```
calls the test404StatusCode() 2 times, but it should call it 4 times.

Expected result: call the test404StatusCode() 4 times with:
```
http://foo/hello
http://www.foo/hello
http://bar/world
http://www.bar/world
```
Actual result:
```
http://foo/hello
http://www.foo/hello
```
It happens due the call of the iterator_to_array() with the second argument equal to `true` in:
```php
namespace PHPUnit\Util;
// ...
class Test
{
// ...
    private static function getDataFromDataProviderAnnotation($docComment, $className, $methodName)
    {
// ...
                if ($data instanceof Traversable) {
                    $data = \iterator_to_array($data);  // <-- here
                }
// ...
   }
```
Possible fix (add a second argument === true):
```php
namespace PHPUnit\Util;
// ...
class Test
{
// ...
    private static function getDataFromDataProviderAnnotation($docComment, $className, $methodName)
    {
// ...
                if ($data instanceof Traversable) {
                    $data = \iterator_to_array($data, false);  // <-- here
                }
// ...
   }
```

**Comment by sebastianbergmann** [comment]
I am confused, maybe because there is a "disconnect" between this issue's title and its description. It's not really about a "second yield from", is it? It's about yielding without (different) keys, right?

Does `\iterator_to_array($data, false)` solve your issue?

**Comment by markbook2** [comment]
> Does \iterator_to_array($data, false) solve your issue?

Yes the `\iterator_to_array($data, false)` solves the issue in this particular case.

**Comment by SimonHeimberg** [comment]
this fix has some backward incompatibility: #2867
